diff -urN freetype-2.4.12.org/src/cff/cffobjs.c freetype-2.4.12/src/cff/cffobjs.c
--- freetype-2.4.12.org/src/cff/cffobjs.c	2013-05-01 11:34:28.000000000 +0200
+++ freetype-2.4.12/src/cff/cffobjs.c	2013-05-09 10:27:13.286264482 +0200
@@ -1056,7 +1056,7 @@
 
 
     /* set default property values */
-    driver->hinting_engine    = FT_CFF_HINTING_FREETYPE;
+    driver->hinting_engine    = FT_CFF_HINTING_ADOBE;
     driver->no_stem_darkening = FALSE;
 
     return FT_Err_Ok;
