--- freetype-2.1.3/include/freetype/config/ftoption.h	Wed Nov  6 23:32:53 2002
+++ freetype-2.1.3.new/include/freetype/config/ftoption.h	Mon Nov 18 13:05:47 2002
@@ -378,7 +378,7 @@
   /*   do not #undef this macro here, since the build system might         */
   /*   define for certain configurations                                   */
   /*                                                                       */
-/* #define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
