--- freetype-2.1.9/ft2demos-2.1.9/Makefile.orig	2004-05-25 14:29:10.000000000 +0200
+++ freetype-2.1.9/ft2demos-2.1.9/Makefile	2004-06-25 21:21:40.190458624 +0200
@@ -122,7 +122,7 @@
   endif
   
   COMMON_LINK = $(LINK) $(subst /,$(COMPILER_SEP),$(COMMON_OBJ))
-  GRAPH_LINK  = $(COMMON_LINK) $(subst /,$(COMPILER_SEP),$(GRAPH_LIB))
+  GRAPH_LINK  = $(COMMON_LINK) $(subst /,$(COMPILER_SEP),$(GRAPH_LIB)) -lm
   GRAPH_LINK2 = $(GRAPH_LINK) $(subst /,$(COMPILER_SEP),$(EXTRA_GRAPH_OBJS))
 
   .PHONY: exes clean distclean
