--- freetype-2.1.1/include/freetype/config/ftoption.h.orig	Sat Jun  8 03:05:56 2002
+++ freetype-2.1.1/include/freetype/config/ftoption.h	Sun Jun 16 11:12:06 2002
@@ -375,7 +375,7 @@
   /* By undefining this, you will only compile the code necessary to load  */
   /* TrueType glyphs without hinting.                                      */
   /*                                                                       */
-#undef  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
