--- freetype-2.7.1/ft2demos-2.7.1/Makefile.orig	2017-01-21 23:38:47.751279050 +0100
+++ freetype-2.7.1/ft2demos-2.7.1/Makefile	2017-01-21 23:40:12.444611419 +0100
@@ -127,10 +127,10 @@
                  $(TOP_DIR)/include \
                  $(SRC_DIR)
 
-  COMPILE = $(CC) $(CPPFLAGS) \
+  COMPILE = $(CC) $(INCLUDES:%=$I%) \
+                  $(CPPFLAGS) \
                   $(CFLAGS) \
                   $(ANSIFLAGS) \
-                  $(INCLUDES:%=$I%) \
                   $DFT_CONFIG_MODULES_H="<ftmodule.h>"
 
   # Enable C99 for gcc to avoid warnings.
