--- ft2demos-2.10.1/Makefile.orig	2019-07-02 17:26:30.444954351 +0200
+++ ft2demos-2.10.1/Makefile	2019-07-02 18:17:59.171554616 +0200
@@ -127,10 +127,10 @@
                  $(TOP_DIR)/include \
                  $(SRC_DIR)
 
-  COMPILE = $(CC) $(CPPFLAGS) \
+  COMPILE = $(CC) $(INCLUDES:%=$I%) \
+                  $(CPPFLAGS) \
                   $(CFLAGS) \
                   $(ANSIFLAGS) \
-                  $(INCLUDES:%=$I%)
 
   # Enable C99 for gcc to avoid warnings.
 
