Index: ftlru.c
===================================================================
RCS file: /cvs/freetype/freetype2/src/cache/ftlru.c,v
retrieving revision 1.27
diff -u -r1.27 ftlru.c
--- ftlru.c	2003/03/20 07:04:40	1.27
+++ ftlru.c	2003/06/07 19:01:36
@@ -231,7 +231,7 @@
   
           if ( node == NULL )
           {
-            FT_ASSERT( list->nodes == 0 );
+            FT_ASSERT( list->num_nodes == 0 );
             error = FT_Err_Out_Of_Memory;
             goto Exit;
           }
@@ -384,6 +384,7 @@
           clazz->node_done( node, list );
 
         FT_FREE( node );
+	list->num_nodes--;
       }
       else
         pnode = &(*pnode)->next;
